File: 1493.py
class Solution:
    def longestSubarray(self, nums: list[int]) -> int:
        i, j, n, k = 0, 0, len(nums), 1
        while j < n:
            if nums[j] == 0:
                k -= 1
            if k < 0:
                if nums[i] == 0:
                    k += 1
                i += 1
            j += 1
        return j - i - 1

if __name__ == '__main__':
    s = Solution()
    nums = list(map(int, input().split()))
    print(s.longestSubarray(nums))

File: cpp/1493.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int i = 0, j = 0, n = nums.size(), k = 1;
        while(j < n){
            if(nums[j] == 0) k--;
            if(k < 0 && nums[i++] == 0) k++;
            j++;
        }
        return j - i - 1;
    }
};

int main(){
    Solution s;
    int n;
    cin >> n;
    vector<int> nums(n);
    for(int i = 0; i < n; i++) 
        cin >> nums[i];
    cout << s.longestSubarray(nums) << endl;
    return 0;
}